<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Promise</title>
</head>
<body>
    <h1>Promise</h1>

    <script>
        const promiseOne = new Promise(function(resolve, reject){
            setTimeout(function() {
                console.log('Async task is complete');
                resolve() //we have to call this resolve function it can't exe below function untill they call.
            }, 3000)
        })

        promiseOne.then(function(){
            console.log('Promise comsumed');
        })

       new Promise(function(resolve, reject) {
            setTimeout(function(){
                console.log('Async task 2');
                resolve()
            },5000)
            
           
        }).then(function() {
               console.log('ji');
            })

        const promiseThree  = new Promise(function(resolve, reject){
            setTimeout(function(){
                console.log('propes three');
                resolve({id:63762, name:'Moinuddin',age:26})
            },1000)
        })
        promiseThree.then(function(users){
            console.log(users);
        })

        const promisFour = new Promise(function(resolve, reject) {
            setTimeout(function(){
                let myErr = true
                if(!myErr){
                    resolve({id:234567, name:'moinuddin'})
                } else{
                    reject('ERRORS: something missing...!')
                }
                
            },1000)
        })
            promisFour.then(function(user){
            console.log(user);
                return user.id
            }).then((username) => {
                console.log(id);
            }).catch((error) => {
                console.log(error);
            }).finally(()=>{
                console.log('The promise is resolved or rejected');
            })


            const promiseFive = new Promise((resolve, reject) => {

                setTimeout(() => {
                        let myErr = false
                    if(!myErr){
                        resolve({id:786567, name:'yaser'})
                    } else{
                        reject('JS: something missing...!')
                    }
                }, 6000)
            })

            async function myFun() {
                try {
                    gotResponse = await promiseFive
                    console.log(gotResponse);
                } catch (error) {
                    console.log(error);
                }
            }

            myFun()
    </script>
</body>
</html>